code/modules/clothing/clothing.dm

```diff
@@ -73,6 +89,8 @@
 		LoadComponent(/datum/component/bloodysoles)
 	if(!icon_state)
 		item_flags |= ABSTRACT
+	if(has_shielding)
+		AddComponent(/datum/component/shielded, max_charges = max_charges, recharge_start_delay = recharge_delay, charge_increment_delay = recharge_rate, charge_recovery = recharge_amount, lose_multiple_charges = lose_multiple_charges, shield_icon = shield_icon)
```

Comment:
not sure if maintainers will like this, kinda goes against the point of components?

Reply:
What's the problem?

Reply:
yeah do go on my friend

Reply:
I’m not sure how that helps, because then the component has to be added in both the current location in /hardsuit/shielded and the new /ctf. I don’t really like the vars either, but it’s a choice between that and duplicating the AddComponent().

Reply:
then duplicate the add component